internal/reconciler/shoot/reconciler.go

```diff
@@ -23,10 +40,125 @@ type Reconciler struct {
+
+	userNameClaim := "sub"
+	groupsClaim := "groups"
+	prefix := buildPrefix(shoot)
+	userNamePrefix := prefix
+	groupsPrefix := prefix
+
+	oidc := emptyOIDC(shoot)
+	if _, err := controllerutils.GetAndCreateOrMergePatch(ctx, r.Client, oidc, func() error {
+		oidc.Annotations = utils.MergeStringMaps(oidc.Annotations, map[string]string{})
+		oidc.Labels = utils.MergeStringMaps(oidc.Labels, map[string]string{
+			LabelKeyManagedBy: LabelValueGardenShootTrustConfigurator,
+		})
+		oidc.Spec = authenticationv1alpha1.OIDCAuthenticationSpec{
+			IssuerURL: issuerURL,
+			// TODO(theoddora): Consider configuring ClientID based on landscape
+			ClientID:       "garden",
+			UsernameClaim:  &userNameClaim,
+			UsernamePrefix: &userNamePrefix,
+			GroupsClaim:    &groupsClaim,
+			GroupsPrefix:   &groupsPrefix,
+		}
+		return nil
+	}); err != nil {
+		log.Error(err, "Failed to create or update OIDC resource", "oidc", client.ObjectKeyFromObject(oidc))
+		return ctrl.Result{RequeueAfter: r.ResyncPeriod}, err
+	}
+
+	log.Info("Successfully created or updated OIDC resource for shoot", "oidc", client.ObjectKeyFromObject(oidc))
+	return ctrl.Result{RequeueAfter: r.ResyncPeriod}, nil
+}
+
+func (r *Reconciler) deleteOIDC(ctx context.Context, log logr.Logger, shoot *gardencorev1beta1.Shoot) (ctrl.Result, error) {
+	oidc := emptyOIDC(shoot)
+	err := r.Client.Get(ctx, types.NamespacedName{Name: getOIDCResourceName(shoot), Namespace: v1beta1constants.GardenNamespace}, oidc)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("OIDC resource not found, nothing to do", "oidc", client.ObjectKeyFromObject(oidc))
+			return reconcile.Result{}, nil
+		}
+		log.Error(err, "Failed to get OIDC resource", "oidc", client.ObjectKeyFromObject(oidc))
+		return reconcile.Result{}, fmt.Errorf("failed to get OIDC: %w", err)
+	}
+
+	if err := r.Client.Delete(ctx, oidc); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("OIDC resource not found, nothing to do", "oidc", client.ObjectKeyFromObject(oidc))
+			return reconcile.Result{}, nil
+		}
+		log.Error(err, "Failed to delete OIDC resource", "oidc", client.ObjectKeyFromObject(oidc))
+		return ctrl.Result{RequeueAfter: r.ResyncPeriod}, fmt.Errorf("failed to delete OIDC: %w", err)
+	}
+	log.Info("Successfully deleted OIDC resource", "oidc", client.ObjectKeyFromObject(oidc))
+	return reconcile.Result{}, nil
+}
+
+func emptyOIDC(shoot *gardencorev1beta1.Shoot) *authenticationv1alpha1.OpenIDConnect {
+	return &authenticationv1alpha1.OpenIDConnect{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      getOIDCResourceName(shoot),
+			Namespace: v1beta1constants.GardenNamespace,
+		},
+	}
+}
+
+func getOIDCResourceName(shoot *gardencorev1beta1.Shoot) string {
+	return fmt.Sprintf("%s-%s", shoot.Namespace, shoot.Name)
+}
 
-	return ctrl.Result{}, nil
+func buildPrefix(shoot *gardencorev1beta1.Shoot) string {
+	return fmt.Sprintf("ns:%s.shoot:%s.id:%s", shoot.Namespace, shoot.Name, string(shoot.UID))
```

Comment:
Several suggestions
- Let's use `:` as the only separator
- We finish the prefix with `:` so that the prefix is separated from the actual username
- We remove the `id` part as it is immediately after the shoot name and it should be clear what that is

```suggestion
	return fmt.Sprintf("ns:%s:shoot:%s:%s:", shoot.Namespace, shoot.Name, string(shoot.UID))
```